internal/transport/http2_server.go

```diff
@@ -464,6 +466,26 @@ func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(
 		})
 		return nil
 	}
+	if headerError != nil {
+		t.controlBuf.put(&earlyAbortStream{
+			httpStatus:     http.StatusBadRequest,
+			streamID:       streamID,
+			contentSubtype: s.contentSubtype,
+			status:         headerError,
+			rst:            !frame.StreamEnded(),
+		})
+		return nil
+	}
+	if !isGRPC {
```

Comment:
It seems to me like this should come first -- why complain about the syntax of some grpc headers if the other side isn't even speaking grpc?

Reply:
Good point! Fixed.